### ISSUE TITLE
Unrecognized symbols should report an error via onError

### ISSUE THREAD

**Issue by LeviSchuck** [TroyAlford/react-jsx-parser#92]
I seem unable to iterate over over bindings in my server side rendering project.

Is there a recommended way to do this?

It fails here in `source/components/JsxParser.js`
```js
      case 'CallExpression':
        const parsedCallee = this.parseExpression(expression.callee)
        if (parsedCallee === undefined) return undefined
        return parsedCallee(...expression.arguments.map(this.parseExpression))
```

While playing around with the min.js version with console.logs, it seems that `expression.arguments.map(this.parseExpression)` evaluates to `undefined`.

Reproducing example:
```tsx
import React from 'react'
import ReactDOMServer from 'react-dom/server'
import JSXParser from 'react-jsx-parser'


const jsx = `
<div>
{data.list.map((entry) => (<span key={entry.id}>{entry.id}</span>))}
</div>
`;
const bindings = {
  data: {
    list: [
      {
        id: 1,
      },
      {
        id: 2,
      },
    ],
  },
};

try {
  console.log("Via JSXParser")
  const result = ReactDOMServer.renderToString(<JSXParser bindings={bindings} jsx={jsx}/>)
  console.log(result);
} catch (e) {
  console.error(e)
}

try {
  console.log("Via native");
  const data = bindings.data;
  const result = ReactDOMServer.renderToString(<div>
    {data.list.map((entry) => (<span key={entry.id}>{entry.id}</span>))}
  </div>)
  console.log(result);
} catch (e) {
  console.error(e);
}
```

```
Via JSXParser
TypeError: Array.prototype.map called on null or undefined
    at map (<anonymous>)
    at e.parseExpression (/home/levi/project/node_modules/react-jsx-parser/lib/webpack:/react-jsx-parser/source/components/JsxParser.js:77:9)
    at props (/home/levi/project/node_modules/react-jsx-parser/lib/webpack:/react-jsx-parser/source/components/JsxParser.js:69:22)
    at Array.map (<anonymous>)
    at e.component [as parseElement] (/home/levi/project/node_modules/react-jsx-parser/lib/webpack:/react-jsx-parser/source/components/JsxParser.js:167:9)
    at Boolean (/home/levi/project/node_modules/react-jsx-parser/lib/webpack:/react-jsx-parser/source/components/JsxParser.js:42:52)
    at Array.map (<anonymous>)
    at e.console [as parseJSX] (/home/levi/project/node_modules/react-jsx-parser/lib/webpack:/react-jsx-parser/source/components/JsxParser.js:37:36)
    at e.trim [as render] (/home/levi/project/node_modules/react-jsx-parser/lib/webpack:/react-jsx-parser/source/components/JsxParser.js:225:40)
    at processChild (/home/levi/project/node_modules/react-dom/cjs/react-dom-server.node.development.js:2959:18)
Via native
<div data-reactroot=""><span>1</span><span>2</span></div>
```

**Comment by LeviSchuck** [comment]
It seems this may be falling under `The component does not support inline function declarations`, as an inline function is used to specify the nested components. 

What then would be a workable solution here?

**Comment by TroyAlford** [comment]
Yes, that's exactly the problem. You can't define the function declaration inside the component. You could do something like this, though:
```
<JSXParser
  bindings={{
    ...bindings,
    renderListItem: entry => <span key={entry.id}>{entry.id}</span>
  }}
  jsx="<div>{data.list.map(renderListItem)}</div>"
/>
```

**Comment by LeviSchuck** [comment]
Could development mode fail more gracefully here, explaining the event rather than resulting in undefined? That or console log when the warnings attribute is set

**Comment by TroyAlford** [comment]
Yeah - the ideal should be an error message of some kind, reported via the `onError` handler, which explains that it encountered an unparsable symbol not found in the bindings, or something similar, and is returning `undefined` as a result.

I would welcome a PR for that. Otherwise, I'll try to get to it sometime soon.

**Comment by TroyAlford** [comment]
This should now be resolved in `v1.19.1` :) Please let me know if you see further problems!

**Comment by LeviSchuck** [comment]
Thank you!
